pom.xml

```diff
@@ -119,25 +120,17 @@ under the License.
       <scope>provided</scope>
     </dependency>
 
-    <!-- Add logging framework, to produce console output when running in the IDE. -->
-    <!-- These dependencies are excluded from the application JAR by default. -->
-    <dependency>
-      <groupId>org.apache.logging.log4j</groupId>
-      <artifactId>log4j-slf4j-impl</artifactId>
-      <version>${log4j.version}</version>
-      <scope>provided</scope>
-    </dependency>
     <dependency>
-      <groupId>org.apache.logging.log4j</groupId>
-      <artifactId>log4j-api</artifactId>
-      <version>${log4j.version}</version>
-      <scope>provided</scope>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+      <version>${slf4j.version}</version>
     </dependency>
+    <!-- Add logging framework, to produce console output when running in the IDE. -->
```

Comment:
can we have the logging change in a separate PR - it is easier to track the history then please

Reply:
Dedicated lib for retries might be an overkill now, but I think we can benefit in long term. The library provides Rate Limiter or Circuit Breaker. Both features might be worth adding. Or at least Rate Limiter.

Reply:
1) Logging lib replacement is a consequence of introducing resilence4j, so I think we should keep this replacement as a part of this PR.

2) I understand Krzysztof's approach to avoid adding dependencies for simple features that can be easily implemented on our own. But in a long term I'm planning to add rate limiter (and possibly circuit breaker), so I think adding resilence4j is acceptable.